### ISSUE TITLE
finit doesn't register input from keyboard

### ISSUE THREAD

**Issue by aifoslinux** [troglobit/finit#82]
From commit 510b53e to 3f15090 finit doesn't register any keys at all, so basically at the login prompt I can not login. I have custom musl based linux distribution using finit as primary init system and I'm happy to say that it works nice and fast. Thank you for the good work!

**Comment by troglobit** [comment]
Hi! Always happy to hear from users of Finit! :smiley: 

Sorry you're experiencing problems.  This sounds like a possible termios and/or udevd/d-bus issue.  At least that's the likely candidates in the range of commits you mention.

How do you start your system, do you have a finit.conf or finit.d/*.conf somewhere I can have a look at? In particular, what getty are you using -- maybe the built-in getty in Finit?  If so, then you could try using an external getty from BusyBox or util-linx, i.e. agetty.

**Comment by aifoslinux** [comment]
I'm mostly using the alpinelinux finit example. You can check this files:
finit.conf:
https://github.com/aifoslinux/rcps/blob/master/infra/finit/finit.cfg
finit.d/tty.conf:
https://github.com/aifoslinux/rcps/blob/master/infra/finit/tty.finitd
recipe for building finit:
https://github.com/aifoslinux/rcps/blob/master/infra/finit/RECIPE

I blieve it could be udev/dbus issue, becuase all the binaries are located at /bin since I'm not using /usr/bin, which is symlink to /.

**Comment by troglobit** [comment]
Hmm, that's odd.  I just double-checked with my Alpine Linux install, which is musl based, and it worked fine.  Maybe it's like you say, a problem with the `udevd` change I did in 34a1524 ... do you think you could try by locally `git revert 34a1524`?  Could be an issue with my having to split the call to start udevd and `udevadm trigger ...`

**Comment by troglobit** [comment]
Another thing of note, in TroglOS I have the following in my kernel .config, maybe not relevant but I mention it anyway just in case:

```
CONFIG_UEVENT_HELPER_PATH="/sbin/mdev"
CONFIG_DEVTMPFS=y
CONFIG_DEVTMPFS_MOUNT=y
```

**Comment by aifoslinux** [comment]
I've only CONFIG_DEVTMPFS=y set, here is the config file:
https://github.com/aifoslinux/rcps/blob/master/infra/linux/config

**Comment by troglobit** [comment]
That should be sufficient, I think.

Noticed now that you're using eudev, or "pure" udev without systemd. Which unfortunately is what I've *not* tested on. Sorry about that! :disappointed: 

Do you think you could try doing a local `git revert` on 34a1524?

**Comment by aifoslinux** [comment]
Could not revert to 34a1524 but the commit before, ie 7a2667d, works fine. It seems to be the udevd issue.

**Comment by aifoslinux** [comment]
I've rebuilt finit with 34a1524 and I have the issue present, so this commit is what causing it.

**Comment by troglobit** [comment]
OK, thank you for helping me track it down!

The reason for the change is that Finit without this change does not monitor `udevd`, so if it crashes it will not be restarted.  

I'll try to install Gentoo in a virtual machine and see if I can figure out what's going wrong.

**Comment by aifoslinux** [comment]
alpinelinux provides eudev so you know :)

**Comment by troglobit** [comment]
It does, huh?  I've just used the BusyBox family of tools and embedded stuff like dropbear, mostly because I'm an embedded geek.  OK, I'll give it a try, thanks! :)

**Comment by troglobit** [comment]
OK, I think I've managed to reproduce your problem now using Void Linux.  Didn't manage to reproduce with Alpine because I wanted to keep mdev.  Anyway, while playing around with Void Linux I discovered they had a workaround for this in their runit implementation, which should also work for Finit.

I'll look into this again tomorrow.  Sorry for taking so long to fix this, but I only have weekends and the odd evenings to work on non-work related Finit use.
